docs/proposals/federated-ruler.md

```diff
@@ -0,0 +1,71 @@
+
+#### Proposal
+
+For federated rules owned by a single tenant, creation of federated rules (those sourcing data from multiple tenants) should be blocked behind the feature flag `tenant-federation.enabled`.
+
+To support composite tenants, if tenant federation is enabled for ruler and alertmanager via `tenant-federation.enabled`, then ruler use a `mergeQueryable` to aggregate the results of querying multiple tenants. The ruler and alertmanager APIs should be updated to always call `tenant.GetTenantIDs` instead of `tenant.GetTenantID`, which will use the MultiTenantResolver when tenant federation is enabled.
+
+### Where to store resulting series of federated rule
+
+#### Challenge
+
+A single tenant rule always stores produced series in the tenant where the rule exists. This 1 -> 1 mapping becomes a many -> 1 mapping for federated rules.
+
+#### Proposal
+
+As some use cases will demand that a specific federated rule, querying tenants `B` and `C`, is stored in the owning team's tenant `A`, an option to allow explicit assignment of destination tenant for a federated rule is needed. Some use cases where a set of tenants `A|B|C|...|Z` are being treated as a single logical tenant when querying this explicit assignment of destination tenant is not needed.
+
+To allow both of these, we suggest an additional field to the rules proto `dest_tenant_id` which determines which tenant to send the series produced by the rule to, used if and only if the rule is a federated rule. If this field is not given, for composite tenants a random but consistent subtenant of the multiple tenants owning the rule is chosen using a hashmod of the series label to determine the subtenant and for single tenants owning a federated rule the resulting series is saved in the tenant which owns the rule.
+
+### Which tenants to query from for federated rules
+
+#### Challenge
+
+A single tenant rule always queries the tenant which owns the rule. This 1 -> 1 mapping becomes a 1 -> many mapping for federated rules.
+
+#### Proposal
+
+As some use cases will demand that a specific federated rule, querying tenant B and C, is stored in the owning teams tenant A, an option to allow explicit assignment of source tenants for a federated rule is needed. In the case of a composite tenant where a set of tenants `A|B|C|...|Z` are being treated as a single logical tenant when querying this explicit assignment of destination tenant isn't explicitly called for, but could prove useful.
+
+To support both of these use cases, we suggest an additional field `source_tenant_ids` on the rule containing an OrgID string e.g. `t0|t1|...|ti` which when present determines which tenants to query for the given rule. In the case of a composite tenant this field would be optional, as ownership of a rule by a composite tenant implies source tenants e.g. for the composite tenant `t0|t1|...|ti` the natural source sub tenants would be `t0`, `t1`, `t2` etc.
+
```

Comment:
Need to add challenge around how do we know a given tenant is allowed to query a set of other tenants and specifically document which endpoints which need to be locked down to prevent a given tenant from querying the world.

Reply:
I've been thinking on this some after the meeting today, and I think that something as simple as a string slice config option like `tenant-federation.ruler-federated-tenant-allow-list` which contains OrgIDs which are permitted to have federated rules e.g. a value of `infra,0|1|2` would allow federated rules for `infra` and `0|1|2`, but not for `0|1` or `2` or `bizz` and so on.

Reply:
i added this so resolving this comment thread